app/uk/gov/hmrc/timetopaytaxpayer/controllers/TaxPayerController.scala

```diff
@@ -16,105 +16,72 @@
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
 import uk.gov.hmrc.timetopaytaxpayer.communication.preferences.CommunicationPreferences
-import uk.gov.hmrc.timetopaytaxpayer.communication.preferences.CommunicationPreferences._
+import uk.gov.hmrc.timetopaytaxpayer.connectors.{DesConnector, SaConnector}
 import uk.gov.hmrc.timetopaytaxpayer.debits.Debits._
-import uk.gov.hmrc.timetopaytaxpayer.infrastructure.DesService.{DesError, DesUnauthorizedError, DesUserNotFoundError}
-import uk.gov.hmrc.timetopaytaxpayer.returns.Returns.{Return, ReturnsResult}
-import uk.gov.hmrc.timetopaytaxpayer.sa.DesignatoryDetails.Individual
-import uk.gov.hmrc.timetopaytaxpayer.sa.SelfAssessmentService.{SaError, SaServiceResult, SaUnauthorizedError, SaUserNotFoundError}
-import uk.gov.hmrc.timetopaytaxpayer.taxpayer.{SelfAssessmentDetails, TaxPayer}
+import uk.gov.hmrc.timetopaytaxpayer.returns.Returns.Return
+import uk.gov.hmrc.timetopaytaxpayer.taxpayer.DesignatoryDetails.Individual
+import uk.gov.hmrc.timetopaytaxpayer.taxpayer.{Interest, SelfAssessmentDetails, TaxPayer}
 import uk.gov.hmrc.timetopaytaxpayer.{AuthorizedUser, Utr, taxpayer}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class TaxPayerController @Inject()(debitsService: (Utr => Future[DebitsResult]),
-                         preferencesService: (Utr => Future[CommunicationPreferencesResult]),
-                         returnsService: (Utr => Future[ReturnsResult]),
-                         saService: ((Utr, AuthorizedUser) => Future[SaServiceResult]),
-                                   cc:ControllerComponents)
-                        (implicit executionContext: ExecutionContext) extends BackendController(cc) {
+class TaxPayerController @Inject() (
+    saConnector:  SaConnector,
+    desConnector: DesConnector,
+    cc:           ControllerComponents)
+  (implicit executionContext: ExecutionContext) extends BackendController(cc) {
 
-  def getTaxPayer(utrAsString: String) = Action.async { implicit request =>
+  def getTaxPayer(utrAsString: String): Action[AnyContent] = Action.async { implicit request =>
     implicit val writeTaxPayer: Writes[TaxPayer] = TaxPayer.writer
 
     val utr = Utr(utrAsString)
 
-    def lookupAuthorizationHeader() = {
-      val headerResult: Either[Result, AuthorizedUser] = request.headers.get(HeaderNames.authorisation).map(AuthorizedUser.apply).toRight(Unauthorized("No authorization header set"))
-      EitherT(Future.successful(headerResult))
-    }
+    val possibleUser: Option[AuthorizedUser] = request.headers.get(HeaderNames.authorisation).map(AuthorizedUser.apply)
 
-    val service: Future[DebitsResult] = debitsService(utr)
-    EitherT(service).leftMap(handleError)
+    possibleUser match {
+      case None => Future.successful(Unauthorized(s"Unauthorized DES call for user with UTR [${utr.value}] not found"))
+      case Some(authorizedUser) => {
+        for {
```

Comment:
Because all connector calls are read only it can be implemented in such way that all features start asap:

```
//start features before for comprehension
val returnsF = desConnector.returns(utr)
val debitsF = desConnector.debits(utr)
val preferencesF = desConnector.preferences(utr)
val individualF = saConnector.individual(utr, authorizedUser)

for {
          returns <- returnsF
          debits <- debitsF
          preferences <- preferencesF
          individual <- individualF
        } yield {
          Ok(Json.toJson(taxPayer(utrAsString, debits, preferences, returns, individual)))
        }
```

Reply:
done